### ISSUE TITLE
Clarification and advice on `scale.factor` in `

### ISSUE THREAD

**Issue by denvercal1234GitHub** [ncborcherding/escape#108]
Hi, Nick, 

I hope all is well and your daughter is recovering well soon. 

When you have a moment, would you mind helping me understand when one should use the `scale.factor` argument in `escape::performNormalization`? 

And, according to https://github.com/ncborcherding/escape/issues/86, this argument `scale.factor` can be set to `scale.factor = SeuratObj$nFeature_RNA` and not a fixed value?

Thank you again for your help!

**Comment by ncborcherding** [comment]
Yes **scale.factor** should be a vector of equal length to the number of cells. If **scale.factor** is NULL - ```performNormalization()``` will calculate a normalization by cell and gene set. 

Warning should be fixed now in the master branch. 

```
devtools::install_github("ncborcherding/escape")
```
Nick

**Comment by denvercal1234GitHub** [comment]
Thanks, Nick. Very helpful. 

One clarification about the heatmap. 

After I ran `escape::performNormalization(make.positive = T, scale.factor = obj$nFeature)`, I then used the result as input to `escape::heatmapEnrichment(assay = "runEscape.ssGSEA_C5_GOBP_normalized"`. 

But, when running `escape::heatmapEnrichment()`, I am still a bit confused why setting `scale=T` here in `escape::heatmapEnrichment()` would return the scores now being again negative? And, when setting `scale=F` in `escape::heatmapEnrichment()`, the values range stays positive.
